### PR TITLE
update Select2 integration example and explanation

### PR DESCRIPTION
**Pull request by Morklympious**
Hey! per #954 I went ahead and modified the docs to have a cleaner, less headache inducing example for integrating select2. 
 
I tried to stray from talking too much about `m.startComputation` and `m.endComputation` like they were what you needed to do. I don't know who authored this section prior, but I also feel like redrawing in the config isn't the devil? 

let me know if there's language/etc that should be changed!

### CODE REVIEW

File: docs/integration.md [hunk 1]
```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
...
-            m.component(Select2, {
-              data: ctrl.data, 
-              value: ctrl.currentUser, 
-              onchange: ctrl.changeUser
-            })
-        ]);
+// Component containing a Select that uses Select2 for functionality.
+var S2Component = {
+
+  controller: function() {
```
Comment (isiahmeadows) [inline docs/integration.md]: This isn't consistent with most of the rest of the docs, so would you mind pulling it out?

File: docs/integration.md [hunk 2]
```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
...
+  }
+}
+
+// Primary component.
+var MainComponent = {
+  controller: function() {
+    var ctrl = this;
+
+    // Putting data in here is arbitrary,
+    // you'd probably get this elsewhere in a normal case.
```
Comment (isiahmeadows) [inline docs/integration.md]: I think it would be okay to just say something along the lines of "some arbitrary data".

File: docs/integration.md [hunk 3]
```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
...
+          {id: 7, name: 'King George'},
+          {id: 8, name: 'Marquis de Lafayette'}
+        ]
+  },
+
+  view: function(ctrl) {
+    return m('div', {class: 'select-container'}, [
+      m('label', 'Historical Figure: '),
+      m(S2Component, {
+        selectedUser: ctrl.selectedUser, // Let's just say Burr is the Default.
```
Comment (isiahmeadows) [inline docs/integration.md]: This comment should really be where `ctrl.selectedUser` is first defined.

File: docs/integration.md [hunk 4]
```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
...
 
-`select2.config` is a factory that creates a `config` function based on a given controller. We declare this outside of the `select2.view` function to avoid cluttering the template.
-
-The `config` function created by our factory only runs the initialization code if it hasn't already. This `if` statement is important, because this function may be called multiple times by Mithril's auto-redrawing system and we don't want to re-initialize select2 at every redraw.
+[Source code in JSFiddle](https://jsfiddle.net/11pz8afy/9/)
 
-The initialization code defines a `change` event handler. Because this handler is not created using Mithril's templating engine (i.e. we're not defining an attribute in a virtual element), we must manually integrate it to the auto-redrawing system.
 
-This can be done by simply calling `m.startComputation` at the beginning, and `m.endComputation` at the end of the function. You must add a pair of these calls for each asynchronous execution thread, unless the thread is already integrated.
+`_config` is a helper function that is called via the `config` attribute in the `select` we render in our `SC2Component.view`
```
Comment (isiahmeadows) [inline docs/integration.md]: `s/_config/_configure/g`, if I'm reading this correctly. :wink:

File: docs/integration.md [hunk 5]
```diff
@@ -131,4 +147,3 @@ m.mount(document.getElementById("widget2-container"), Widget2)
 
 m.endComputation()
 ```
-
```
Comment (isiahmeadows) [inline docs/integration.md]: I'm guessing this was your editor misbehaving (i.e. not honoring the [EditorConfig](http://editorconfig.org/) here)?

File: docs/integration.md [hunk 6]
```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
...
+    if(!(jQuery && jQuery.fn.select2))
+      return console.error('You need jquery and Select2 in the page')
+
+    // jQuery element
+    var lmnt = $(element);
+
+    // If this hasn't been initialized, we can do our setup  
+    if(!initialized) {
+
+       lmnt.select2({
```
Comment (isiahmeadows) [inline docs/integration.md]: Try using this instead:

```js
if (!initialized) {
  $(element).select2({
    // whatever
  })
}
```

It's a little more idiomatic for jQuery, since it's a one-time use.

File: docs/integration.md [hunk 7]
```diff
@@ -7,110 +7,126 @@ It's recommended that you encapsulate integration code in a component or a helpe
...
+      the function at the config property in attrs is passed the real DOM element
+      (element), and whether or not it exists in the page (initialized)
+
+      Anyways, let's look at using Select2 now.
+    */
+
+
+    // Ensure Externals exist, otherwise we'll error out.
+    if(!(jQuery && jQuery.fn.select2))
+      return console.error('You need jquery and Select2 in the page')
```
Comment (isiahmeadows) [inline docs/integration.md]: For purposes of demonstration, I don't think this is necessary.

**Comment by isiahmeadows** [comment]
@Morklympious 

I'm taking this one step at a time. The prose feels really awkward to read, almost as much as what was there previously, no offense.

Also, while you're at it, mind using tabs instead of spaces for the code? That's the general style used across the project.

**Comment by Morklympious** [comment]
No offense taken. I have a pitfall of being needlessly verbose >.>

I appreciate the feedback! I'll get back to this when I have some time! :)

**Comment by syaiful6** [comment]
see here #1094

**Comment by isiahmeadows** [comment]
@syaiful6 I like this better, BTW, since it's also fixing the awkward prose that went with the example.

@Morklympious Haven't completely forgotten about it ;-)

**Comment by pygy** [comment]
@Morklympious @isiahmeadows Is this ready to merge, beside the conflict and white space issue?

**Comment by isiahmeadows** [comment]
@pygy Not exactly. The code is okay (mod whitespace and comments), but the prose is only a small improvement over the original. (I *was* going to work with him to make it better, but he's mostly disappeared.)

If you want to take a stab at this, go for it.

**Comment by Morklympious** [comment]
Oh I'm still totally here! I was wondering what you'd like to see in terms of prose but I never really heard anything so I guess this thread just kinda got stale :P

**Comment by Morklympious** [comment]
I'll make another pass at it in the next few days and see where I can remove fluff! For reference, is it the comments or the code that are hard or uncomfortable to digest?

**Comment by isiahmeadows** [comment]
@Morklympious Your last comment here was last April (about 4 1/2 months ago), and I pinged you a week later, but no response since. That's where I was coming from. 😉

**Comment by isiahmeadows** [comment]
(Oh, and you probably have some pretty nasty merge conflicts to address, BTW.)

**Comment by Morklympious** [comment]
Might be better for me to salvage the example and then just re-fork from master :P 

At least that way I can make sure I have EditorConfig in place!

**Comment by isiahmeadows** [comment]
Yeah. It might be best to start from scratch at this point.

**Comment by Morklympious** [comment]
Sorry about the lack of response! I'll make sure to keep things alive this time :)

**Comment by isiahmeadows** [comment]
Okay! Ping me once the new PR is set up.

**Comment by Morklympious** [comment]
As it turns out, pulling and rebasing all the changes made since this PR was opened was painless, so I think this PR can stay open and serve as the PR. 

two questions:

1) Would you rather I open this in a fresh PR and just close this?
2) I was going to see if there was anything else I could do after I removed some wordy prose, but I think this might be a good point for a spot check. 

I re-read what I wrote and you're right. I dawdle. I dawdle a lot. So I wanted to take the most salient stuff from the example as it exists, and sort of make the language conversational so it doesn't feel too hard to parse for any skill level. 

Throw whatever feedback you got! Even if it's EditorConfig or style related. pretty sure my EC should be working. 

Again, sorry about my MASSIVE delay and short-term disappearance!

**Comment by tivac** [comment]
Quick scan and I don't see anything that should keep this from landing at this point. Vote via GH reactions and I can merge this later tonight.

**Comment by pygy** [comment]
LGTM content-wise, you may want to squash the history (you can do it from the [github interface](https://github.com/blog/2141-squash-your-commits)).

**Comment by Morklympious** [comment]
Shamelessly upvoting myself.

**Comment by tivac** [comment]
:tada: merged :tada:

**Comment by isiahmeadows** [comment]
PSA: Updated references to @stffrd's old username upon private request.
